sql/2023/06/ttfb-by-language.sql

```diff
@@ -0,0 +1,70 @@
+# HTTP Archive query to get TTFB information of WordPress sites by language.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See https://github.com/GoogleChromeLabs/wpp-research/pull/54
+
+SELECT
+  client,
+  lang,
+  COUNT(DISTINCT origin) AS n,
+  SUM(
+    IF
+      (ttfb.start < 200, ttfb.density, 0)) / SUM(ttfb.density) AS fast,
+  SUM(
+    IF
+      (ttfb.start >= 200
+         AND ttfb.start < 1000, ttfb.density, 0)) / SUM(ttfb.density) AS avg,
+  SUM(
+    IF
+      (ttfb.start >= 1000, ttfb.density, 0)) / SUM(ttfb.density) AS slow
+FROM
+  `chrome-ux-report.all.202304`,
```

Comment:
Not sure about this, do you have a reference query where you took this bit from? I'm not saying it's incorrect, but I've typically seen CWV queries use the `chrome-ux-report.materialized.device_summary` which already include `fast_ttfb`, `avg_ttfb`, and `slow_ttfb` as fields for each origin, so more efficient and less error-prone to query that rather than having logic in here to calculate it.

That approach is e.g. what the CWV Tech Report uses: https://github.com/HTTPArchive/cwv-tech-report/blob/main/sql/monthly.sql

Simple example:
```sql
CREATE TEMP FUNCTION IS_GOOD(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
  SAFE_DIVIDE(good, good + needs_improvement + poor) >= 0.75
);

CREATE TEMP FUNCTION IS_NON_ZERO(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
  good + needs_improvement + poor > 0
);

SELECT
  date,
  device,
  origin,
  IS_NON_ZERO(fast_ttfb, avg_ttfb, slow_ttfb) AS any_ttfb,
  IS_GOOD(fast_ttfb, avg_ttfb, slow_ttfb) AS good_ttfb
FROM
  `chrome-ux-report.materialized.device_summary`
WHERE
  date = CAST("2023-05-01" AS DATE)
```

Reply:
I think I originally got it from https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2022/performance/ttfb_by_cms.sql

I'm all for simplifying though 👍